src/communityid.bif

```diff
@@ -49,8 +49,8 @@ function community_id_v1%(cid: conn_id, seed: count &default=0, do_base64: bool
 
             break;
         case TRANSPORT_UNKNOWN:
-            emit_builtin_error("CommunityID: unknown transport layer", cid);
-            return zeek::make_intrusive<zeek::StringVal>("");
+            hash_proto = ntohs(orig_port->Port());
```

Comment:
Is this left-over from the initial version using orig_port->Port() to encode the proto? Could use cid's proto field now?

I think we also need to ignore the ports in the `digest_update()` calls below according to the [spec](https://github.com/corelight/community-id-spec).

Reply:
> Is this left-over from the initial version using orig_port->Port() to encode the proto? Could use cid's proto field now?

Fixing this fixed the `FAIL`s you mentioned below.

> I think we also need to ignore the ports in the digest_update() calls below according to the [spec](https://github.com/corelight/community-id-spec).

Agreed that it's not matching the spec, but I'll fix that separate from this PR as it's out of scope.

Reply:
The new approach [here](https://github.com/zeek/zeek/pull/3950/files#diff-788f371ac608aa4080cb2be2d0e6374b34a69b10482f8c4c95ca14322d7f0eecR59) casting down `cid_proto` to 8 bits strikes me as right. We have to keep the algorithm compatible because it's not just about old vs new IDs, it's also about IDs computed elsewhere on the same traffic. Also, the 8-bit value will be correct — I can't imagine Community ID will ever encounter in-kernel values like that 262 for MPTCP.